includes/class-freemius.php

```diff
@@ -4034,42 +4034,49 @@ private function should_run_connectivity_test( $flush_if_no_connectivity = false
         }
 
         /**
-         * @author Vova Feldman (@svovaf)
-         * @since  1.1.7.4
+         * @author Leo Fajardo (@leorw)
+         * @since  2.5.3
          *
-         * @param int|null $blog_id      Since 2.0.0.
-         * @param bool     $is_gdpr_test Since 2.0.2. Perform only the GDPR test.
+         * @param bool $is_update
          *
-         * @return object|false
+         * @return bool
          */
-        private function ping( $blog_id = null, $is_gdpr_test = false ) {
-            if ( WP_FS__SIMULATE_NO_API_CONNECTIVITY ) {
-                return false;
+        private function should_turn_fs_on( $is_update = true ) {
+            if ( empty( $this->_plugin->moderation ) ) {
+                return true;
             }
 
-            $version = $this->get_plugin_version();
+            $optin_config = $this->_plugin->moderation['opt_in'];

```

Comment:
@fajardoleo we can't assume that the developer correctly uses the flag, so we need to validate `$this->_plugin->moderation['opt_in']` is not empty before accessing it.